source/MaterialXCore/Definition.cpp

```diff
@@ -10,6 +10,38 @@
 namespace MaterialX
 {
 
+namespace
+{
+    InterfaceElementPtr findMatchingImplementation(const NodeDef& nodedef, const string& target, const vector<InterfaceElementPtr>& candidates)
+    {
+        // Search for the first implementation which matches the given target string.
+        for (InterfaceElementPtr interface : candidates)
+        {
+            if (!interface->isA<Implementation>() ||
+                !targetStringsMatch(interface->getTarget(), target) ||
+                !nodedef.isVersionCompatible(interface))
+            {
+                continue;
+            }
+            return interface;
+        }
+
+        // Search for a node graph match if no implementation match was found.
```

Comment:
This split into two loops was done in the old code, so I didn't change that. I think it's because we consider source code or native implementations to have higher priority than node graph implementations. And I think that makes sense, a native or source code implementation is more specific so should have higher priority.

Reply:
Ok. I misread the diff then. Thanks.